runtime/drivers/duckdb/connectors.go

```diff
@@ -53,14 +60,20 @@ func (c *connection) ingestFile(ctx context.Context, env *connectors.Env, source
 		path = filepath.Join(env.RepoDSN, path)
 	}
 
+	// get all files in case glob passed
+	localPaths, err := doublestar.FilepathGlob(path)
+	if err != nil {
+		return err
+	}
+
 	// Not using query args since not quite sure about behaviour of injecting table names that way.
 	// Also, it's a source, so the caller can be trusted.
 
 	var from string
 	if conf.Format == ".csv" && conf.CSVDelimiter != "" {
-		from = fmt.Sprintf("read_csv_auto('%s', delim='%s')", path, conf.CSVDelimiter)
+		from = fmt.Sprintf("read_csv_auto(['%s'], delim='%s')", path, conf.CSVDelimiter)
 	} else {
-		from, err = getSourceReader(path)
+		from, err = getSourceReader(localPaths)
 		if err != nil {
 			return err
 		}
```

Comment:
`conf.Format` is just based on the path, not `fileutil.FullExt(paths[0])` as in `getSourceReader`. This can lead to the CSV delimiter not propagating when the glob pattern doesn't end in `.csv` as you pointed out. Maybe simplify and pass `CSVDelimiter` into `getSourceReader` instead?

Reply:
In that case, it could actually just call `ingestFiles` after expanding the glob with `doublestar`, right?

Reply:
Yes, but the `ParseConfig` func doesn't do that when it sets `conf.Format`. Hence, it's best not to rely on it.